blob/blob.go

```diff
@@ -768,8 +772,6 @@ func (b *Bucket) NewWriter(ctx context.Context, key string, opts *WriterOptions)
 		end:        end,
 		cancel:     cancel,
 		key:        key,
-		opts:       dopts,
```

Comment:
These two fields (along with `ctx`) are only supposed to be set when we need to call `NewTypedWriter` later; they are explicitly set in the `else` a few lines down. I'm not sure why we were setting them here, they are not needed and it made the code unclear because the comment in the struct says they are only set when `w` is `nil`.

I removed them and clarified the comments.